weaviate/collections/tenants.py

```diff
@@ -128,27 +138,71 @@ def get(self) -> Dict[str, Tenant]:
             `weaviate.UnexpectedStatusCodeError`
                 If Weaviate reports a non-OK status.
         """
-        path = "/schema/" + self.__name + "/tenants"
-        response = self.__connection.get(
-            path=path,
-            error_msg=f"Could not get collection tenants for {self.__name}",
-            status_codes=_ExpectedStatusCodes(
-                ok_in=200, error=f"Get collection tenants for {self.__name}"
-            ),
-        )
+        if self.__connection._weaviate_version.supports_tenants_get_grpc:
+            return self.__get_with_grpc()
+        else:
+            return self.__get_with_rest()
 
-        tenant_resp: List[Dict[str, Any]] = response.json()
-        return {tenant["name"]: Tenant(**tenant) for tenant in tenant_resp}
+    def get_by_names(self, names: Sequence[str]) -> Dict[str, Tenant]:
+        """Return named tenants currently associated with a collection in Weaviate.
+
+        If the tenant does not exist, it will not be included in the response.
+
+        If no names are provided, all tenants will be returned.
+
+        The collection must have been created with multi-tenancy enabled.
+
+        Arguments:
+            `names`
+                Optional list of tenant names to retrieve. If not provided, all tenants will be returned.
+
+        Raises:
+            `weaviate.WeaviateConnectionError`
+                If the network connection to Weaviate fails.
+            `weaviate.UnexpectedStatusCodeError`
+                If Weaviate reports a non-OK status.
+        """
+        self.__connection._weaviate_version.check_is_at_least_1_25_0("The 'get_by_names' method")
+        if self.__validate_arguments:
+            _validate_input(
+                [_ValidateArgument(expected=[Sequence[str]], name="names", value=names)]
+            )
+        return self.__get_with_grpc(names=names)
 
-    def update(self, tenants: Union[Tenant, Sequence[Tenant]]) -> None:
+    def get_by_name(self, name: str) -> Optional[Tenant]:
```

Comment:
can't we do one function with Union[str, List[str]] and type overloading?

Reply:
Yes, nice idea!

Reply:
Hmm this doesn't play with the type checking. `pyright` complains that `str` and `Sequence[str]` overlap. In addition, we'd have a `get_by_names` endpoint that could return `Tenant` and `None`, which might be unexpected for non-type-hinting users. I'd prefer to keep the two methods separate due to this, wdyt?